apiserver/facades/agent/uniter/networkinfoiaas.go

```diff
@@ -273,35 +300,30 @@ func (n *NetworkInfoIAAS) populateUnitMachine() error {
 	return nil
 }
 
-// populateMachineNetworkInfos sets network info for the unit's machine
+// populateMachineAddresses sets addresses for the unit's machine
 // based on devices with addresses in the unit's bound spaces.
-func (n *NetworkInfoIAAS) populateMachineNetworkInfos() error {
+func (n *NetworkInfoIAAS) populateMachineAddresses() error {
 	spaceSet := set.NewStrings()
 	for _, binding := range n.bindings {
 		spaceSet.Add(binding)
 	}
 
-	var privateIPAddress string
-	n.machineNetworkInfos = make(map[string]params.NetworkInfoResult)
+	var privateMachineAddress network.SpaceAddress
+	n.machineAddresses = make(map[string][]NetInfoAddress)
 
 	if spaceSet.Contains(network.AlphaSpaceId) {
 		var err error
-		privateMachineAddress, err := n.pollForAddress(n.machine.PrivateAddress)
+		privateMachineAddress, err = n.pollForAddress(n.machine.PrivateAddress)
 		if err != nil {
-			n.machineNetworkInfos[network.AlphaSpaceId] = params.NetworkInfoResult{Error: apiservererrors.ServerError(
```

Comment:
These errors were pointless anyway, because they never surfaced. `network-get` just reported that there were no addresses for the binding. At least they are logged now.